### DISCUSSION TITLE
EV AC Charger Support

### DISCUSSION THREAD

**Discussion by fbradyirl** [TypQxQ/Sigenergy-Home-Assistant-Integration#2] [Ideas]
This is a really great project. I installed yesterday and it works great so thank you!

I also have the AC charger installed and was wondering if any of those sensors would be available via Modbus?

**Comment by TypQxQ** [comment]
Hello!

I haven't found any exposed sensors for the AC charger in the documentation.
The values for DC charger do not work for me and I commented the m out.
Maybe they only work if you have a DC charger and maybe they also work with an AC charger.
You can search the yaml file for "charger" and uncomment them to try.

**Reply by fbradyirl** [reply]
Yes good points. I will try and let you know.

**Reply by Barrstard** [reply]
If it helps, I have the DC charger and uncommented the code
The only thing I have issues with is the output power but the code also mentioned this was an issue.
Current also holds the last values if the charge has stopped. Works a treat when it is running though
Created an additional page in the PV dashboard to show it

![image](https://github.com/user-attachments/assets/f5992cf2-c47b-4138-b5fa-2e95a06709fd)

**Reply by runn8** [reply]
Today, I received a message from Sigenergy regarding the missing 'Sigen DC Charger - Output Power.' They said that the address 31502 worked for them in the test version SPC105 (?). I will keep you updated. ;)

**Reply by DotOnedotNL** [reply]
Output power is a uint32 instead of a uint16 which is in the config.
Change it to uint32 and it works like a charm

**Reply by runn8** [reply]
Yehaa, thank you very much!

**Reply by runn8** [reply]
In addition you need to comment out "scale: 0.1". Otherwise only a tenth of the value is displayed.

**Reply by DeepSpace7** [reply]
Here is the updated MODBUS documentation including the EVAC. I would also love to have the data available, but don't know how to so :-(
[Sigenergy Modbus Protocol V2.3 - 20241209.pdf](https://github.com/user-attachments/files/19757715/Sigenergy.Modbus.Protocol.V2.3.-.20241209.pdf)

**Reply by stevenh** [reply]
There's already v2.5 elsewhere in this thread.

**Comment by Hammi71** [comment]
Hi there !

I'm very pleased for having found this project today. My SigEnergy - equipment is similar to Andrei's. Additionally I own the AC Charger and the Gateway (https://www.sigenergy.com/en/products/gateway). I'm running a HomeServer with TrueNAS Scale, but haven't yet installed HomeAssistant on it, because TrueNAS Scale will move from Kubernetes to Docker Compose soon (https://forums.truenas.com/t/the-future-of-electric-eel-and-apps/5409). So it's likely to become a project for upcoming autumn/winter including the SigEnergy-integration.

You might be interested to know that SigEnergy has released version 1.8 of their ModBus protocol : https://support.sigenergy.com/cn/notice-details?noticeId=2003&lang=en
The version description says : "Added AC-Charger model and it’s related registers. Added AC-Charger related system state and alarm appendix."

Best regards from germany,
      Marcus

**Reply by TypQxQ** [reply]
Thank you!
How did you find this, it's not listed publicly. Wondering so I can update with the next update.
I will integrate this as soon as I get the time (Depends on the kids).

**Reply by Hammi71** [reply]
Hi Andrei !
After logging in under https://support.sigenergy.com/cn/main?lang=en (top right corner) with my standard enduser-account, a new submenu "Notification" appeared on the left with two subsubmenus "Technical" and "Product". Among other things I found "SigEnergy Modbus Protocol" under "Technical".
I hope, that helps.
Best regards,
Marcus

**Reply by fbradyirl** [reply]
Yes and note you probably want a new secret to represent the different slave ID as it won't be the same as the inverter slave ID.

**Reply by fbradyirl** [reply]
> Hi Andrei ! After logging in under https://support.sigenergy.com/cn/main?lang=en (top right corner) with my standard enduser-account, a new submenu "Notification" appeared on the right with two subsubmenus "Technical" and "Product". Among other things I found "SigEnergy Modbus Protocol" under "Technical". I hope, that helps. Best regards, Marcus

Seems this document is no longer on their site.

**Reply by Hammi71** [reply]
Yes. I can confirm. The mentioned submenu "Notification" has disappeared using my credentials.

**Reply by TypQxQ** [reply]
Did you manage to maybe download the document? 
Was it a newer version?

**Reply by fbradyirl** [reply]
V1.8 is the latest I had downloaded.

**Reply by TypQxQ** [reply]
Thank you, that is the same as I have 😄

**Reply by Hammi71** [reply]
I found the file in Windows-Trashcan on my PC at work and attached it below.

[Sigenergy Modbus Protocol - 20240724.pdf](https://github.com/user-attachments/files/17395543/Sigenergy.Modbus.Protocol.-.20240724.pdf)

**Reply by mattgrech** [reply]
@TypQxQ @fbradyirl @purcell-lab , I just received the latest version of the Modbus Protocol from AU Tech V2.5 which has a lot of changes for the AC charger both Read only and  parameter setting updates. See attached if not already[](url)
[Modbus Protocol EN - SIGEN_v2.5.pdf](https://github.com/user-attachments/files/19727504/Modbus.Protocol.EN.-.SIGEN_v2.5.pdf)

**Reply by slampt** [reply]
Do we need to request updates to the AC Charger via the installer?
My AC charger was installed by a local electrician after my System was installed.

**Reply by TypQxQ** [reply]
Thank you for the heads up.
Everything but the AC and DC charger registers are already updated.

AC and DC chargers are coming in the HACS implementation I'm working on. DC is confirmed reads the registers but write is untested yet. AC is untested.
I don't own neither so can't test them.

If you have a spare Home Assistant installation I welcome you to test it. Wouldn't replace the current YAML implementation for production just yet.
/Andrei

**Reply by mattgrech** [reply]
@TypQxQ i have the AC charger and happy to test although i only have the one home assistant instance.

**Reply by alexeiw123** [reply]
> Thank you for the heads up. Everything but the AC and DC charger registers are already updated.
> 
> AC and DC chargers are coming in the HACS implementation I'm working on. DC is confirmed reads the registers but write is untested yet. AC is untested. I don't own neither so can't test them.
> 
> If you have a spare Home Assistant installation I welcome you to test it. Wouldn't replace the current YAML implementation for production just yet. /Andrei

I've been trying to test using HACS integration - on my main HA install alongside the yaml method (yolo).

I can set up the main sigen plant OK. AC charger can be 'added' by I don't get any entity updates. I suspect despite having the test firmware on AC EV charger, the modus isn't enabled (there's no option like with the main plant in the installer app). I've asked support to enable modbus, so see where we get to.

**Reply by mattgrech** [reply]
@TypQxQ i've setup a test instance of my HA and added your latest HACS integration. It automatically discovers 3 new sig devices with different IP's. One is for the inverter, one is for the AC charger and the 3rd i'm not sure what its for. In any case it only allows me to add the Inverter and not the others.

**Reply by alexeiw123** [reply]
> @TypQxQ i've setup a test instance of my HA and added your latest HACS integration. It automatically discovers 3 new sig devices with different IP's. One is for the inverter, one is for the AC charger and the 3rd i'm not sure what its for. In any case it only allows me to add the Inverter and not the others.

My DHCP server shows separate IPs for inverter, gateway and ACEV charger, so I'd suggest the third is the gateway.

Where do you see the discovered IPs through the integration? Mine doesn't show my ACEV charger anywhere.

**Reply by mattgrech** [reply]
@alexeiw123 
![image](https://github.com/user-attachments/assets/1d5a69eb-f240-44d9-a366-75bbb33f9ae8)

**Reply by alexeiw123** [reply]
> @alexeiw123 

Ah, I haven't had any auto-discovery from this integration. I just saw there was another recent update in HACS (v0.3.5-alpha) with DHCP discovery in the changelog. I've installed this and see if I get any auto-discovery.

**Comment by fbradyirl** [comment]
Thank you. I am not having much luck with those, but I suspect I need my installer to enable Modbus also on the AC EV Charger with slave ID 2.

**Comment by Hammi71** [comment]
Could you provide us with a short feedback (maybe some screenshots ?), if enabling ModBus-TCP on the AC Charger solved your issue ?
Regards,
Marcus

**Reply by fbradyirl** [reply]
Yes I have asked my installer to enable it. I don't have the installer access myself but I will confirm if it works.

**Comment by TypQxQ** [comment]
Allright!
I have made a [branch](https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/tree/ac-charger_development) that has the values read from modbus of AC Charger. Please also set up the new sigen_ac_modbus_slave parameter in secrets.yaml
I don't have this charger, already had another when bought my Sigenergy plant, so I can't test it.
I need som help interpreting this values:
- Sigen AC Charger - System state
  System states according to IEC61851-1 definition:
  I haven't found the exact definition of the states. The values are just a guess:
  0. System innit: A state where the EVSE is not connected to the vehicle and is not ready to supply energy to the vehicle.???
  1. A1/A2: A state where the EVSE is connected to the vehicle and is ready to supply energy to the vehicle.???
  2. B1: A state where the EVSE is supplying energy to the vehicle.???
  3. B2: A state where the EVSE is connected to the vehicle and is ready to supply energy to the vehicle, but the vehicle is not connected.???
  4. C1: A state where the EVSE is connected to the vehicle and is ready to supply energy to the vehicle, but the vehicle is not ready to receive energy.???
  6. C2: Not a clue
  7. F: Fault???
  8. E: Error???

- Sigen AC Charger - Charging power
  [6, X] X is the smaller value between the rated current and the AC-Charger input breaker rated current.
  I need some examples.
  
Thank you all for helping out!
Regards,
Andrei

**Reply by fbradyirl** [reply]
Thank you @TypQxQ 

I've been trying different things but cannot get anything to work. I'm working with Sigenergy support to get it going. I think either Modbus is not enabled or there is a slave ID conflict. Will report back here as I progress...

**Reply by darwinzhou7** [reply]
@TypQxQ Is it possible for this project to query multiple Sigenergy inverters with different slave addresses?
if so, I suggest @fbradyirl to simply change Modbus slave address of the Sigenergy invert's to another number, instead of changing AC charger's slave address

best regards from China

**Reply by fbradyirl** [reply]
Yes I already tried this by trying slave address 2 with the inverter IP but it didn't work. Sigenergy R&D showed me a screenshot of the backend showing slave address of 1 for the AC Charger but I fear that conflicts with the inverter so I'm awaiting them to change it. Even installer does not see the Modbus setting for the AC Charger in his app.

**Reply by fbradyirl** [reply]
Update: I got a call from Sigenergy support team and they said AC Charger Modbus firmware is not released yet, but should be out before the end of Sept and probably before then. When it gets updated, I will test again.

**Reply by JuergenOrth** [reply]
Hi, is the new FW already available? I didn't notice any change.

**Reply by fbradyirl** [reply]
Haven't seen any sign of it yet.

**Reply by JuergenOrth** [reply]
FW of my AC Charger has changed from V100R001C10B03**1B** to V100R001C10B03**5A**. According to my installer there are no modbus settings in the installer app for the charger. So I tried some addresses (1, 2, 3, 4) but no success :-(

**Reply by TypQxQ** [reply]
We'll just have to wait a bit more for Sigenergy...

**Reply by Janneck79** [reply]
Has anybody tried to use OCPP for the charger? That could work. I will get my instalation in the next couple of days and will try it with the OCPP integration in Home Assistant. As i know you can enable OCPP in the my sigen app.

**Reply by JuergenOrth** [reply]
Yes you can enable OCPP in the user app. Let us know about your success.
![grafik](https://github.com/user-attachments/assets/48059d8e-4e2c-4c86-8fc6-953e67e5930a)

**Reply by fbradyirl** [reply]
Doesn’t let you type in the server though. Only 1 value selectable. I tried hooking it up to the OCPO addon server in HA but you cannot enter the server url.

**Reply by alexeiw123** [reply]
> We'll just have to wait a bit more for Sigenergy...

Is this still the current state of AC EV charger metrics? My sig system was commissioned today - all the battery details in home assistant thanks to this, just wondering if the AC part is working before I dive in to making more changes.

As above, I also couldn't configure the OCPP server either.

**Reply by alexeiw123** [reply]
For the EV charger, do we still use the same `sigen_modbus_host_ip:`?
I can see 3x sigen mac addresses in my router. One is the inverter, which I'm using for this integration. I assume the other two are the gateway and the AC EV charger? Do we need to query the AC EV charger directly or can we expect to get its values from the inverter?

**Comment by fbradyirl** [comment]
Doesn’t let you type in the server though. Only 1 value selectable. On 27 Nov 2024, at 15:58, JuergenOrth ***@***.***> wrote:﻿
Yes you can enable OCPP in the user app. Let us know about your success.
grafik.png (view on web)

**Comment by sejonr73** [comment]
I Also realy need to have AC charger implementation. I tryed to find AC charger with modbus reader program but i could not find it?? Help needed please!

**Reply by fbradyirl** [reply]
Same. I havent been able to get it working either via Modbus.

**Reply by JuergenOrth** [reply]
AFAIK the AC charger doesn't support Modbus yet. Even it is documented otherwise in the modbus API. And there is no way to setup a modbus slave address with the installer app right now. So we have to wait ...
As I can see in my installation the only component listening on default modbus port is the inverter.
$ nmap -p 502-502 -T4 -A sigec | grep 502   # Inverter
502/tcp open  tcpwrapped

$ nmap -p 502-502 -T4 -A sigwb | grep 502  # Wallbox / AC Charger
502/tcp closed mbap

**Comment by stevenh** [comment]
I can confirm the AC charger doesn't seem to be available, checked all slaveid's on the inverter and direct to EV charger neither work. Have raised a ticket with our installer to check with Sigenergy what's needed.

**Comment by stevenh** [comment]
I've been doing some digging on the AC charger state and have found the info.

This is what I have in local branch, which also splits up the components into separate files for easier management: `ac_charger_system_state.yaml`
```yaml
# System states according to IEC61851-1 definition:
#
# The values are based IEC 61851-1:2017 as documented by:
# https://www.mathworks.com/help/autoblks/ug/charge-an-electric-vehicle.html

0: System init - System is initialising # Not part of the IEC 61851-1 standard
1: A1/A2 - Vehicle is not connected # A1: Charger not ready, A2: Charger ready
2: B1 - Vehicle connected and not ready to accept energy, Charger not ready to supply energy
3: B2 - Vehicle connected and not ready to accept energy, Charger ready to supply energy
4: C1 - Vehicle connected and ready to accept energy, EV does not require charging area ventilation, Charger not ready to supply energy
5: C2 - Vehicle connected and ready to accept energy, EV does not require charging area ventilation, Charger ready to supply energy
6: F - Fault Other Charger problem (can be intentionally set by the Charger, for example, that maintenance is required)
7: E - Error Charger disconnected from vehicle / Charger disconnected from utility, Charger loss of utility power or control pilot short to control pilot reference
```

I can upload my work in progress branch as PR if folks are interested.

**Comment by stevenh** [comment]
Folks can find my take in this draft PR #62, it breaks the config up into components allowing it to be more easily managed and extended.

**Reply by alexeiw123** [reply]
looks fantastic - just to check, does this now function, or is this preparatory work from the documentation for when modbus from the AC charger is accessible?

**Reply by stevenh** [reply]
preparation work, I've tested locally and as mentioned the AC charger doesn't seem to respond currently. I have a ticket in with my supplier to touch base with Sigenergy to get an update on this.

**Comment by alexeiw123** [comment]
Thankyou, I'll keep watching this space.

I've also been in touch with the Sig software team because my AC charger
has been wildly missing the mark on its solar tracking performance.

They have told me that the modbus update for AC EV charger isn't ready yet,
but no details when. They are working to get compatibility with OCPP using
home assistant though, so hopefully that's soon and can at least enable
some local viability and control.

**Reply by Janneck79** [reply]
> Thankyou, I'll keep watching this space. I've also been in touch with the Sig software team because my AC charger has been wildly missing the mark on its solar tracking performance. They have told me that the modbus update for AC EV charger isn't ready yet, but no details when. They are working to get compatibility with OCPP using home assistant though, so hopefully that's soon and can at least enable some local viability and control.
> […](#)
> On Fri, 14 Feb 2025, 11:02 pm Steven Hartland, ***@***.***> wrote: preparation work, I've tested locally and as mentioned the AC charger doesn't seem to respond currently. I have a ticket in with my supplier to touch base with Sigenergy to get an update on this. — Reply to this email directly, view it on GitHub <[#2 (reply in thread)](https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2#discussioncomment-12200001)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ASRURBPV7R5TOKQUQ2P6SNL2PXLMLAVCNFSM6AAAAABNVGG6USVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMRQGAYDAMI> . You are receiving this because you commented.Message ID: <TypQxQ/Sigenergy-Home-Assistant-Integration/repo-discussions/2/comments/12200001 @github.com>

Have you got any reply? If i try to solar charge with the AC charger it charges with 1.4 kW or 4.4kW but nothing in between. I am expecting that it will follow the solar power to use all of it instad of feeding the grid.

**Reply by stevenh** [reply]
The response I got yesterday was that the AC Charger modbus support isn't implemented yet. So it may be a while until we see movement on this unfortunately

**Reply by stevenh** [reply]
> Have you got any reply? If i try to solar charge with the AC charger it charges with 1.4 kW or 4.4kW but nothing in between. I am expecting that it will follow the solar power to use all of it instad of feeding the grid. 

I'm not sure but the steps could be something that is negotiated between the charger and the EV? It's hard to tell for sure as there's no historical graphs in the Sigen app so without modbus support I can't see the data here

**Reply by alexeiw123** [reply]
> > Thankyou, I'll keep watching this space. I've also been in touch with the Sig software team because my AC charger has been wildly missing the mark on its solar tracking performance. They have told me that the modbus update for AC EV charger isn't ready yet, but no details when. They are working to get compatibility with OCPP using home assistant though, so hopefully that's soon and can at least enable some local viability and control.
> > […](#)
> > On Fri, 14 Feb 2025, 11:02 pm Steven Hartland, _**@**_.***> wrote: preparation work, I've tested locally and as mentioned the AC charger doesn't seem to respond currently. I have a ticket in with my supplier to touch base with Sigenergy to get an update on this. — Reply to this email directly, view it on GitHub <[#2 (reply in thread)](https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2#discussioncomment-12200001)>, or unsubscribe https://github.com/notifications/unsubscribe-auth/ASRURBPV7R5TOKQUQ2P6SNL2PXLMLAVCNFSM6AAAAABNVGG6USVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMRQGAYDAMI . You are receiving this because you commented.Message ID: <TypQxQ/Sigenergy-Home-Assistant-Integration/repo-discussions/2/comments/12200001 @github.com>
> 
> Have you got any reply? If i try to solar charge with the AC charger it charges with 1.4 kW or 4.4kW but nothing in between. I am expecting that it will follow the solar power to use all of it instad of feeding the grid.

This is basically the issue I have too. I have a long email conversation still on gong with sig to understand and solve this issue, also the lack of modbus and ocpp support. Their intentions to fix this are good but in all honesty I'm frustrated with the AC charger not doing what it is advertised to do.

There's two issues that contribute to what you're seeing - one is that 3 phase switching does not work. It will switch to 3 phase and then not switch back. If you turn that feature off, you will have more stability. The other issue is that 100% PV mode doesn't respond in useful time periods. I've found a bit more luck with the dynamic load management option turned on (which I don't understand, compared to 100% pv mode without it?)

In any case, I have a gen 3 tesla wall charger on the other side of my garage, so I just use that and the tesla api to control charging of my car, which is really annoying.

**Reply by stevenh** [reply]
On the 100% PV charging, we hit that today and our installer said it prefers to charge the batteries vs EV if they need it. You can work around that by caping the max state of charge below its current value which will force the excess to be used by the EV charger.

**Reply by alexeiw123** [reply]
If you change to solar boost mode - then the ev charger takes priority over the sigenstore. At least that's how they explained it to me, I haven't tested it out. It's easier to just plug in to the tesla charger, as I know exactly what it will do and I can see it in home assistant.

**Reply by stevenh** [reply]
I tested that yesterday too, same behaviour as 100% PV so batteries still took priority

**Comment by slampt** [comment]
Just got my AC Charger installed today. So far 100% PV doesn't work at all for me, would never charge due to no green energy. I have set it to Solar Boost Charging with 0kW from grid. I was hoping I could grab the power consumption from this and add it to HA etc. But I am guessing we are going to have to wait for them to allow Modbus and/or OCPP to HASS.
Appreciate all the work y'all have done thank you so much!

**Comment by fbradyirl** [comment]
<img width="453" alt="image" src="https://github.com/user-attachments/assets/831a8667-5a67-433b-8aab-7e6e25ebb2f6" />

I have received beta firmware for the AC charger from Sigenergy and can confirm modbus is working now for me.

<img width="2040" alt="Screenshot 2025-03-19 at 10 04 50" src="https://github.com/user-attachments/assets/661a74ba-2f07-4ed2-9103-3d1f36776ab8" />

I was able to start and stop charging. 

Start by writing 0 to 42000 and stop by writing 1.

<img width="1102" alt="image" src="https://github.com/user-attachments/assets/97c06ee5-987d-4ea3-bb0a-9353ff80237b" />

Some notes:
* You address the inverter IP address directly.
* The AC charger is just another slave. Sig support set that to `2` for me. I assume this will be in the installer app? 
* I set this in secrets.yaml: `sigen_ac_modbus_slave: 2`
* I'm not sure why the **Sigen AC Charger - Start/Stop code** is `65535`. Can anyone comment?

I'll open a PR.

**Reply by fbradyirl** [reply]
PR Open: 

https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2

**Reply by purcell-lab** [reply]
Great news!

Are you able to change the charging_amps? To ramp up and down the charging rate

Does the charger also support OCPP or just modbus for monitoring and control?

**Reply by fbradyirl** [reply]
I haven't had success in changing the amps im afraid but maybe im doing something wrong. 

<img width="1043" alt="image" src="https://github.com/user-attachments/assets/a77f68e9-e10d-4b26-b064-2e2d7110ce14" />

**Reply by purcell-lab** [reply]
> I haven't had success in changing the amps im afraid but maybe im doing something wrong.

The charging_amp  controls are likely to be another modbus register that Sig will need to advise.

**Reply by darwinzhou7** [reply]
reason for Sigen AC Charger - Start/Stop code is 65535, is that this register is Write Only.  If you read it, according to modbus protocol, returns FF FF in Hex,  and that is 65535 in Dec.

**Reply by Janneck79** [reply]
Hi, can you provide your yaml part for the charger? Thanks.

**Reply by TypQxQ** [reply]
There is no workin yaml code for ac charger.
Working code is for the new integration that uses HACS. 
https://github.com/TypQxQ/Sigenergy-Local-Modbus

**Comment by TypQxQ** [comment]
If you feel brave, you can try out the new HACS integration I'm working on.
I don't have a charger so I can't test that functionality:
https://github.com/TypQxQ/HACS-Sigenergy-Local-Modbus

**Reply by alexeiw123** [reply]
I am feeling brave - I can get my sigenstore to work correctly using this integration, however I can't get AC EV charger to appear.
I have been working with sigen support to get many issues with the device resolved and they've pushed me on to a test firmware.

e.g. I now have options for battery/ac ev charger priority. I don't have any specific options for modbus in either the user or the installer app (like we see in sigenstore).

When adding the AC EV charger in the integration, do you use the IP address of the sigenstore, or of the AC EV charger? by using the AC ev charger IP, I get a device added but no entity data comes through. Oddly, when I ping the AC EV charger IP, I don't get any response.

**Reply by fbradyirl** [reply]
See https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2#discussioncomment-12549624

Sigen support or your installer need to set the slave ID for the AC charger. I got them to set it to 2. 

You don't use the AC charger IP address. Everything goes via the inverter IP address for modbus.

**Reply by TypQxQ** [reply]
That's good to know! I thought it used it's own ip address and only the DC charger used the inverters ip address.
I appreciate all the info you give me 😄
I don't have any charger to test with.

**Reply by alexeiw123** [reply]
> See [#2 (comment)](https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2#discussioncomment-12549624)
> 
> Sigen support or your installer need to set the slave ID for the AC charger. I got them to set it to 2.
> 
> You don't use the AC charger IP address. Everything goes via the inverter IP address for modbus.

Thanks, I'll have another play with it soon. In the end, who set the slave ID? Sigen or your installer? My installer and I were not able to find any modbus settings at all under the AC EV charger (I'm on version V100R001C10SPC107B040A).

I've been trying to test using the new integration rather than the yaml approach, so I don't go breaking what is already working for me! I'll focus on using the inverter IP now though. Thanks.

**Reply by JuergenOrth** [reply]
Ultimately, it doesn't matter how the Modbus address of the AC Charges is set if you know what it is. To do this, you could simply test the possible addresses with a small script. At https://www.modbusdriver.com you could download the modpoll CLI-Tool for example. It's free.
To read out the "photovoltaic power"  for example you can use it like this:
`pi@pipv:~/bin $ modpoll -0 -1 -a 247 -r 30035 -t 3:int -i 192.168.3.170
modpoll 3.11 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright (c) 2002-2024 proconX Pty Ltd
Visit https://www.modbusdriver.com for Modbus libraries and tools.

Protocol configuration: MODBUS/TCP, FC4
Slave configuration...: address = 247, start reference = 30035 (PDU), count = 1
Communication.........: 192.168.3.170, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 32-bit integer, input register table
Word swapping.........: Slave configured as big-endian word machine

-- Polling slave...
[30035]: 1955
`

**Reply by fbradyirl** [reply]
> > See [#2 (comment)](https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/discussions/2#discussioncomment-12549624)
> 
> > 
> 
> > Sigen support or your installer need to set the slave ID for the AC charger. I got them to set it to 2.
> 
> > 
> 
> > You don't use the AC charger IP address. Everything goes via the inverter IP address for modbus.
> 
> 
> 
> Thanks, I'll have another play with it soon. In the end, who set the slave ID? Sigen or your installer? My installer and I were not able to find any modbus settings at all under the AC EV charger (I'm on version V100R001C10SPC107B040A).
> 
> 
> 
> I've been trying to test using the new integration rather than the yaml approach, so I don't go breaking what is already working for me! I'll focus on using the inverter IP now though. Thanks.

It was Sigen support who set it for me. Perhaps the installer needs a new app version before they can set it and since the firmware is beta the app version hasn't caught up yet.

**Reply by alexeiw123** [reply]
Sigen have enabled modbus on my AC charger and your HACS integration is working fantastically.

ACEV charger on slave 1 and sigenstore on slave 2.

**Reply by mattgrech** [reply]
> Sigen have enabled modbus on my AC charger and your HACS integration is working fantastically.
> 
> ACEV charger on slave 1 and sigenstore on slave 2.

Can you adjust the amps from HA?

**Reply by alexeiw123** [reply]
I haven't got that far yet - maybe tomorrow. I've just noticed I'm using the old repository since TypQxQ had to change the name. So tomorrow I will remove the yaml version, re-set up with the latest integration and see what works/doesn't.

TypQxQ doesn't have an AC EV charger so has built support blind, I'd be amazed if every feature worked correctly in such an early Dev state.

@TypQxQ I'm not very Dev inclined but would be happy to help you debug features. Mine is the 3 phase version. I could set up a HA login or wireguard VPN access in a private conversation.

**Reply by mattgrech** [reply]
@alexeiw123 thats ok, I’ve been testing the new integration :) 

unfortunately adjusting amps is not working and have been working with @TypQxQ to troubleshoot and resolve

**Reply by DeepSpace7** [reply]
> Sigen have enabled modbus on my AC charger and your HACS integration is working fantastically.
> 
> ACEV charger on slave 1 and sigenstore on slave 2.

How did you contact them? Via support ticket or any direct contact?

**Reply by alexeiw123** [reply]
Emails - that have been ongoing for months.

**Reply by DeepSpace7** [reply]
Now it works for me as well after my inverter has been set to 2. :-)

**Comment by BeauGiles** [comment]
Tried the new HACS version - I assume we need the VIO0R001C10SPC107B040A firmware?
I wasn't prompted / the AC charger didn't show as a device.

(currently running V100R001C10SPC106B035L)

**Reply by BeauGiles** [reply]
Disregard, I missed the part about having to go back and add another device.

**Reply by fbradyirl** [reply]
> Disregard, I missed the part about having to go back and add another device.

Let us know if it works for you.

**Comment by slampt** [comment]
This is rad, these things don't appear to have a physical button to unlock charging and I hate having to use the app all the time hopefully when modbus is released to the masses I can use a somrig to unlock it. Just slap it on the side of the charger :P

**Reply by alexeiw123** [reply]
I took a tesla lead off a tesla wall charger and wired it up - 7m instead of 5m and you get the button that stops charging and unlocks the port 👌just had to find 3.3V inside the Sig ac charger to power it (hint: the LED board).

**Reply by slampt** [reply]
Oh my that is amazing.

**Comment by cjungde** [comment]
You might have gotten the same notification: 

EVAC got a new SW Version today: V100R001C10SPC108B042 with many new Wallbox Features ([LINK](https://www.sigenergy.com/en/support/learn?open_node=304,441&selected_node=571))

**Reply by cjungde** [reply]
Interesting:
I requested to get via a ticket a slaveID set for the EVAC
They set it to the value 2.
After this the slaveID 1 for the inverter was not accessible anymore, so i testet to set the secret.yaml value to: 'sigen_modbus_slave: 2 '

Then the inverter access worked again and i am getting access to the EVAC as well.
For the others who enabled the EVAC as well: Do you have a different slaveID for the Inverter and another one for the EVAC ?

**Comment by TypQxQ** [comment]
I´ve put what I know about it in the new integrations FAQ.
https://github.com/TypQxQ/Sigenergy-Local-Modbus/discussions/37

**Comment by TypQxQ** [comment]
As an update. Charger support will not be added to this integration. 
it is confirmed working on the new one and this one will be phased out as it is hard to push out updates to all users.
New integration at:
https://github.com/TypQxQ/Sigenergy-Local-Modbus

**Reply by Janneck79** [reply]
If i use the new Version, what happens to the history, Dashboard etc. Is the naming of the sensors the same?

**Reply by TypQxQ** [reply]
No, sensor naming isn't the same. It had to change to acomodate multiple devices.
History is not saved unfortunatley
I wish Home Assistant would allow to migrate history to another entity but it doesn't.

**Comment by se-wen** [comment]
I got an new Firmware Update with an maybe interesting changelog:

![Screenshot_20250607-232338](https://github.com/user-attachments/assets/27c89a39-46be-4615-8aed-99558bc9c5d5)

**Reply by TypQxQ** [reply]
That sounds interesting. I don't have any info about it though.

**Comment by vdvmichel** [comment]
What is the latest version of the modbus documentation?

**Reply by TypQxQ** [reply]
2.5

**Comment by DmytroRybka** [comment]
Any news on this ? I was considering having this system but now not sure it's worse it.

**Reply by TypQxQ** [reply]
Developed for the new version. 

https://github.com/TypQxQ/Sigenergy-Local-Modbus

**Reply by erwinsanders** [reply]
Hello 
i have installed this https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration  can i also install  https://github.com/TypQxQ/Sigenergy-Local-Modbus o?

**Reply by TypQxQ** [reply]
No, you have to remove the old one before adding the new.

**Comment by YpsilonTM** [comment]
Question here anyone here have the TCP Modbus enabled on a standalone charger. Because I messaged support for this and all they said is TCP Modbus is not possible for a stand alone charger. Although I am not quite convinced this is the truth. So atm the moment I am furious with this product. The software is terrible and normally I would fix that by controlling it myself. But I can't enable the modbus myself. So my "smart" charger is actually just a dumb expensive charger.

**Comment by erwinsanders** [comment]
I am using evcc for controling my ac charger
To make it smarter

**Reply by YpsilonTM** [reply]
But for that I need TCP Modbus enabled. And sigen won't enable it because they tell me its not possible if you have just the charger alone.
